config/config/src/main/java/io/helidon/config/spi/AbstractMpSource.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.config.spi;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.helidon.common.CollectionsHelper;
+import io.helidon.common.reactive.Flow;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+/**
+ * MP Config source basis. Just extend this class to be both Helidon config source and an MP config source.
+ * @param <S> Type of the stamp of this config source
```

Comment:
Can `S` be, say, `Object`?  Or is there some subtype we could use instead here?

Reply:
There is no supertype shared. I have checked the implementations, and we have `Instant`, `byte[]` (and maybe others).
I have not modified this type, it was there on `AbstractSource` since the beginning.